akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowFoldAsyncSpec.scala

```diff
@@ -261,6 +258,14 @@ class FlowFoldAsyncSpec extends StreamSpec {
 
       upstream.expectCancellation()
     }
+
+    "complete future and return zero given an empty stream" in assertAllStagesStopped {
+      val futureValue =
+        Source.fromIterator[Int](() ⇒ Iterator.empty)
+          .runFoldAsync(0)((acc, elem) ⇒ Future.successful(acc + elem))
+
+      Await.result(futureValue, 3.seconds) should be(0)
```

Comment:
`remainingOrDefault` is a good choice instead of `3.seconds` (it will currently be 3.seconds)